test_integration/test_template.py.in

```diff
@@ -0,0 +1,154 @@
+        to pass to the test-application for the next run.  This is
+        especially useful for tests that execute the test-application
+        multiple times.
+
+        """
+        return []
+
+
+class TestIntegration_@test_name@(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Create launcher, execute benchmark and set up class variables.
+
+        """
+        sys.stdout.write('(' + os.path.basename(__file__).split('.')[0] +
+                         '.' + cls.__name__ + ') ...')
+        test_name = '@test_name@'
+        cls._report_path = test_name + '.report'
+        cls._trace_path = test_name + '.trace'
+        cls._skip_launch = _g_skip_launch
+        cls._keep_files = os.getenv('GEOPM_KEEP_FILES') is not None
+        cls._agent_conf_path = test_name + '-agent-config.json'
+        # Clear out exception record for python 2 support
+        if 'exc_clear' in dir(sys):
+            sys.exc_clear()
+        if not cls._skip_launch:
+            # Set the job size parameters
+            num_node = 1
+            num_rank = 1
+            time_limit = 6000
+            # Configure the test application
+            app_conf = AppConf()
+            # Configure the agent
+            # Query for the min and sticker frequency and run the
+            # energy efficient agent over this range.
+            freq_min = geopm_test_launcher.geopmread("CPUINFO::FREQ_MIN board 0")
+            freq_sticker = geopm_test_launcher.geopmread("CPUINFO::FREQ_STICKER board 0")
+            agent_conf_dict = {'FREQ_MIN':freq_min,
+                               'FREQ_MAX':freq_sticker}
+            agent_conf = geopmpy.io.AgentConf(cls._agent_conf_path,
+                                              'energy_efficient',
+                                              agent_conf_dict)
+            # Create the test launcher with the above configuration
+            launcher = geopm_test_launcher.TestLauncher(app_conf,
+                                                        agent_conf,
+                                                        cls._report_path,
+                                                        cls._trace_path,
+                                                        time_limit=time_limit)
+            launcher.set_num_node(num_node)
+            launcher.set_num_rank(num_rank)
+            # Run the test application
+            launcher.run(test_name)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean up any files that may have been created during the test if we
+        are not handling an exception and the GEOPM_KEEP_FILES
+        environment variable is unset.
+
+        """
+
+        if (sys.exc_info() == (None, None, None) and not
+            cls._keep_files and not
+            cls._skip_launch):
+            os.unlink(cls._agent_conf_path)
+            os.unlink(cls._report_path)
+            for tf in glob.glob(cls._trace_path + '.*'):
+                os.unlink(tf)
```

Comment:
it would be nice if this logic was available in a common place too (maybe not for this PR)

Reply:
Yeah, we need to have a common module for integration tests. Will not fix this now.